### ISSUE TITLE
Updating Windsor to support Castle.Core@5.0.0 and modern TFMs

### ISSUE THREAD

**Issue by Jevonius** [castleproject/Windsor#612]
With the release of Castle.Core@5.0.0, Windsor needs updating to support it as there are breaking changes (some detail in #610). 5.0.0 also has an updated list of TFMs:
```
<TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0</TargetFrameworks>
```

The current targets for Windsor are:
```
Castle.Facilities.AspNet.Mvc = net45
Castle.Facilities.AspNet.Mvc.Tests = net45
Castle.Facilities.AspNet.SystemWeb = net45
Castle.Facilities.AspNet.SystemWeb.Tests = net45
Castle.Facilities.AspNet.WebApi = net45
Castle.Facilities.AspNet.WebApi.Tests = net45
Castle.Facilities.AspNetCore = netstandard2.0
Castle.Facilities.AspNetCore.Tests = netcoreapp2.0
Castle.Facilities.Logging = net45; netstandard2.0
Castle.Facilities.WcfIntegration = net45
Castle.Facilities.WcfIntegration.Demo = net45
Castle.Facilities.WcfIntegration.Tests = net45
Castle.Windsor = net45; netstandard2.0
Castle.Windsor.Extensions.DependencyInjection = netstandard2.0
Castle.Windsor.Extensions.DependencyInjection.Tests = net5.0; 
Castle.Windsor.Tests = net45; net5.0
```

Having not done much with the Windsor codebase, is it insane to suggest aligning with the Core TFM list, where reasonable? Does targeting `net6.0` even make sense for Windsor? At the very least, it should support `net462;netstandard2.0` against Castle.Core@5.0.0, which might be a good first step.

Could also consider retiring some of the Facilities, download counts don't always set the world on fire (https://www.nuget.org/packages/Castle.Facilities.AspNet.Mvc/, https://www.nuget.org/packages/Castle.Facilities.AspNet.WebApi/).

**Comment by jonorossi** [comment]
> is it insane to suggest aligning with the Core TFM list, where reasonable? Does targeting `net6.0` even make sense for Windsor? At the very least, it should support `net462;netstandard2.0` against Castle.Core@5.0.0, which might be a good first step.

Haha, I just finished reviewing your PR and asked the same question. Happy for it to be a 2 stage process. At this point I don't see much point adding more support for .NET Standard if no one has a reason for it, adding .NET6 is probably the way to go moving forward if .NET Standard is taking a back seat.

> Could also consider retiring some of the Facilities

Best to keep this issue to the one topic, it allows people following the repo to track changes easier. Happy to discuss those 2 in another issue.

**Comment by rsr-maersk** [comment]
Any luck here with a new build here. Latest Moq gives:
![image](https://user-images.githubusercontent.com/76153851/173004238-d9af25b9-51d9-4d3d-aaaa-5d5a6a625ab5.png)
on integration tests with:
`_container = new WindsorContainer();`

**Comment by tasoss** [comment]
Is there any way to help with this one? @Jevonius

**Comment by stakx** [comment]
@Jevonius, I've only really done work on DynamicProxy; I hardly know Windsor at all. So I'm afraid I'm not in the best position to help out here. That being said, I csn probably take an hour or two to review any PRs if that would be helpful at all.

**Comment by tasoss** [comment]
Ok, so only @jonorossi could help...
Waiting for news from him.

**Comment by jonorossi** [comment]
@Jevonius I coded reviewed the pull request the day after you created it, it looks like you made further commits afterwards and further comments with the last one starting "Haven't had a chance to get back into the code for this". I assume it's ready to re-review and merge?

**Comment by Jevonius** [comment]
@jonorossi apologies, I did push some further commits, but didn't comment in that thread so hadn't twigged you wouldn't magically see that it was ready to go 😆 I did request another review later on, but that might have slipped through the cracks.

So, #613 is ready to review, I'll make a comment there to be clearer.

I'm doing the `net6.0` changes separately, if you have any thoughts on #616 that would be helpful - I'm not sure as this point whether to just drop the Facility support, or try and make it work with a bodge :)

**Comment by Jevonius** [comment]
@tasoss - coincidentally I was cleaning up some [very] old tabs/windows the other day and one of them was https://github.com/NancyFx/Nancy.Bootstrappers.Windsor/issues/57 - I'd been having the same issue. Small world 😄

**Comment by tasoss** [comment]
@Jevonius @jonorossi thanks for closing this issue.
@Jevonius  hahahahahaha.Indeed!

**Comment by tasoss** [comment]
What are we waiting for v6.0.0 ?
Aren't #611 and #604 fixed already? @jonorossi

**Comment by haldiggs** [comment]
upgraded to Castle.Core 5.0.0, Castle.Windsor 5.1.1 and getting the Could not load type 'Castle.Core.Internal.PermissionUtil' error seen by so many. Cannot upgrade Castle.Windsor 5.1.2 where they say this might be fixed because the requirement says Castle.Core < 5.0.0 ... I guess all I can do is downgrade and wait for 6.0

**Comment by jonorossi** [comment]
@haldiggs Why is NuGet letting you install incompatible versions?

**Comment by jonorossi** [comment]
@haldiggs This is exactly why we released 5.1.2. NuGet doesn't care about a major version change and just assumes it'll be compatible. Yes, you'll need to use Castle Core < 5.0.0.
